### PR TITLE
Bug 1948778: Handle carriage return in Rsync logs for better accuracy

### PR DESCRIPTION
**Pull request by pranavgaikwad**

### CODE REVIEW

**Review by alaypatel07 — APPROVED** [review]
ACK

**Comment by pranavgaikwad** [comment]
The GH action issue is causing the CI to fail even today. I have opened a ticket and would like to wait for the reply from support folks before merging this PR.

**Pull request merged**
